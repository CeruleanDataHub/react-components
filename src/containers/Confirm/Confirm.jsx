import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import ModalWrapper from "styled-react-modal";

import { Button } from "../Button/Button";
import { Typography } from "../Typography/Typography";

const StyledModal = ModalWrapper.styled`
  display: flex;
  background-color: white;
  flex-direction: column;
  padding: 1rem;
  box-shadow: 0 0 18px -3px rgba(27, 27, 27, 0.8);
  width: 40%;
`;

const TextAlignRight = styled.div`
  text-align: right;

  button {
    margin: 0.2rem;
  }
`;

const ConfirmButton = styled.button`
  background: blue;
  color: white;
  font-weight: bold;
`;

const CancelButton = styled.button`
  color: blue;
  text-decoration: underline;
`;

export const Confirm = ({ title, content, isOpen, onConfirm, onCancel }) => (
  <StyledModal isOpen={isOpen} data-styled-modal-test>
    <Typography fontFamily="openSans">
      <div>{title}</div>

      <div>{content}</div>

      <TextAlignRight>
        <Button
          color="transparent"
          onClick={onCancel}
          as={CancelButton}
          data-cancel-button-test
        >
          Cancel
        </Button>

        <Button onClick={onConfirm} as={ConfirmButton} data-confirm-button-test>
          Confirm
        </Button>
      </TextAlignRight>
    </Typography>
  </StyledModal>
);

Confirm.propTypes = {
  title: PropTypes.string,
  content: PropTypes.node,
  isOpen: PropTypes.bool,
  onConfirm: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired
};

Confirm.defaultProps = {
  title: "",
  content: "",
  isOpen: false
};
