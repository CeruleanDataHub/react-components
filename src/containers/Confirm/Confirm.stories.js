import React, { useState } from "react";

import { Button } from "../Button/Button";
import { Typography } from "../Typography/Typography";
import { Confirm } from "./Confirm";

export default {
  title: "Confirm",
  component: Confirm
};

const ConfirmToggle = () => {
  const [isOpen, setIsOpen] = useState(false);

  const toggleConfirm = () => {
    setIsOpen(!isOpen);
  };

  return (
    <Typography fontFamily="openSans">
      <Button onClick={toggleConfirm} color="blue">
        Open Confirm
      </Button>
      <Confirm
        title="Confirm title"
        text="Confirm modal content"
        isOpen={isOpen}
        onConfirm={toggleConfirm}
        onCancel={toggleConfirm}
      />
    </Typography>
  );
};

export const ConfirmToggleStory = () => (
  <div style={{ minHeight: 500, maxWidth: 600 }}>
    <ConfirmToggle />
  </div>
);
