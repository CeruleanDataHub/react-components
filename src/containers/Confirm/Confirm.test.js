import { mount } from "enzyme";
import React from "react";
import { ModalProvider } from "styled-react-modal";

import { Confirm } from "./Confirm";

describe("Confirm", () => {
  let component;
  let handleOnCancelMock;
  let handleOnConfirmMock;

  beforeEach(() => {
    handleOnCancelMock = jest.fn();
    handleOnConfirmMock = jest.fn();

    component = mount(
      <ModalProvider>
        <Confirm
          title="Some title"
          content="Some modal content"
          isOpen
          onCancel={handleOnCancelMock}
          onConfirm={handleOnConfirmMock}
        />
      </ModalProvider>
    );
  });

  it("renders", () => {
    expect(component).toMatchSnapshot();
  });

  it("does not call onCancel yet", () => {
    expect(handleOnCancelMock).not.toHaveBeenCalled();
  });

  it("when cancel button is clicked, calls onCancel", () => {
    component
      .find("ForwardRef[data-cancel-button-test]")
      .props()
      .onClick();

    expect(handleOnCancelMock).toHaveBeenCalled();
  });

  it("does not call onConfirm yet", () => {
    expect(handleOnConfirmMock).not.toHaveBeenCalled();
  });

  it("when confirm button is called, calls onConfirm", () => {
    component
      .find("ForwardRef[data-confirm-button-test]")
      .props()
      .onClick();

    expect(handleOnConfirmMock).toHaveBeenCalled();
  });

  it("given isOpen is false, should not be open", () => {
    component = mount(
      <ModalProvider>
        <Confirm isOpen={false} onConfirm={() => {}} onCancel={() => {}} />
      </ModalProvider>
    );

    expect(component.find("div[data-styled-modal-test]")).not.toExist();
  });
});
