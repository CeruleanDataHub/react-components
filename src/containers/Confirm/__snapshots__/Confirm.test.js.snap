// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm renders 1`] = `
<p>
  <Confirm
    content="Some modal content"
    isOpen={true}
    onCancel={[MockFunction]}
    onConfirm={[MockFunction]}
    title="Some title"
  >
    <Component
      data-styled-modal-test={true}
      isOpen={true}
    >
      <j
        WrapperComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "baseHash": 2127463992,
              "componentId": "Modal__wrap-gmoXzh",
              "isStatic": false,
              "rules": Array [
                "
  display: flex;
  background-color: white;
  flex-direction: column;
  padding: 1rem;
  box-shadow: 0 0 18px -3px rgba(27, 27, 27, 0.8);
  width: 40%;
",
              ],
              "staticRulesId": "",
            },
            "displayName": "Modal__wrap",
            "foldedComponentIds": Array [],
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "Modal__wrap-gmoXzh",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        backgroundProps={Object {}}
        data-styled-modal-test={true}
        isOpen={true}
      >
        <Portal
          containerInfo={
            <div>
              <div
                class="baseStyles__BaseModalBackground-sTbLb hGVCXQ"
              >
                <div
                  class="Modal__wrap-gmoXzh gWaikR"
                  data-styled-modal-test="true"
                >
                  <div
                    class="sc-AxhUy iumSJR"
                    font-family="openSans"
                  >
                    <div>
                      Some title
                    </div>
                    <div>
                      Some modal content
                    </div>
                    <div
                      class="sc-AxgMl jrJXms"
                    >
                      <button
                        class="sc-Axmtr fGzfKN sc-AxiKw cMQRsw"
                        color="transparent"
                        data-button-test="true"
                        type="button"
                      >
                        
                        Cancel
                      </button>
                      <button
                        class="sc-AxheI bgVlsT sc-AxiKw Rtuhc"
                        color=""
                        data-button-test="true"
                        type="button"
                      >
                        
                        Confirm
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          }
        >
          <baseStyles__BaseModalBackground
            onClick={[Function]}
          >
            <div
              className="baseStyles__BaseModalBackground-sTbLb hGVCXQ"
              onClick={[Function]}
            >
              <Modal__wrap
                data-styled-modal-test={true}
              >
                <div
                  className="Modal__wrap-gmoXzh gWaikR"
                  data-styled-modal-test={true}
                >
                  <Typography
                    color=""
                    fontFamily="openSans"
                    size="normal"
                  >
                    <styled.div
                      fontFamily="openSans"
                      size="normal"
                      textColor=""
                    >
                      <div
                        className="sc-AxhUy iumSJR"
                        fontFamily="openSans"
                        size="normal"
                      >
                        <div>
                          Some title
                        </div>
                        <div>
                          Some modal content
                        </div>
                        <styled.div>
                          <div
                            className="sc-AxgMl jrJXms"
                          >
                            <ForwardRef
                              as={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "baseHash": -1474970742,
                                    "componentId": "sc-Axmtr",
                                    "isStatic": false,
                                    "rules": Array [
                                      "
  color: blue;
  text-decoration: underline;
",
                                    ],
                                    "staticRulesId": "",
                                  },
                                  "displayName": "styled.button",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "shouldForwardProp": undefined,
                                  "styledComponentId": "sc-Axmtr",
                                  "target": "button",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              className=""
                              color="transparent"
                              data-cancel-button-test={true}
                              icon=""
                              onClick={[MockFunction]}
                              type="button"
                            >
                              <styled.button
                                as={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "baseHash": -1474970742,
                                      "componentId": "sc-Axmtr",
                                      "isStatic": false,
                                      "rules": Array [
                                        "
  color: blue;
  text-decoration: underline;
",
                                      ],
                                      "staticRulesId": "",
                                    },
                                    "displayName": "styled.button",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "shouldForwardProp": undefined,
                                    "styledComponentId": "sc-Axmtr",
                                    "target": "button",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                className=""
                                color="transparent"
                                data-button-test={true}
                                onClick={[MockFunction]}
                                type="button"
                              >
                                <styled.button
                                  className="sc-AxiKw cMQRsw"
                                  color="transparent"
                                  data-button-test={true}
                                  onClick={[MockFunction]}
                                  type="button"
                                >
                                  <button
                                    className="sc-Axmtr fGzfKN sc-AxiKw cMQRsw"
                                    color="transparent"
                                    data-button-test={true}
                                    onClick={[MockFunction]}
                                    type="button"
                                  >
                                    Cancel
                                  </button>
                                </styled.button>
                              </styled.button>
                            </ForwardRef>
                            <ForwardRef
                              as={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "baseHash": 1882630949,
                                    "componentId": "sc-AxheI",
                                    "isStatic": false,
                                    "rules": Array [
                                      "
  background: blue;
  color: white;
  font-weight: bold;
",
                                    ],
                                    "staticRulesId": "",
                                  },
                                  "displayName": "styled.button",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "shouldForwardProp": undefined,
                                  "styledComponentId": "sc-AxheI",
                                  "target": "button",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              className=""
                              color=""
                              data-confirm-button-test={true}
                              icon=""
                              onClick={[MockFunction]}
                              type="button"
                            >
                              <styled.button
                                as={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "baseHash": 1882630949,
                                      "componentId": "sc-AxheI",
                                      "isStatic": false,
                                      "rules": Array [
                                        "
  background: blue;
  color: white;
  font-weight: bold;
",
                                      ],
                                      "staticRulesId": "",
                                    },
                                    "displayName": "styled.button",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "shouldForwardProp": undefined,
                                    "styledComponentId": "sc-AxheI",
                                    "target": "button",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                className=""
                                color=""
                                data-button-test={true}
                                onClick={[MockFunction]}
                                type="button"
                              >
                                <styled.button
                                  className="sc-AxiKw Rtuhc"
                                  color=""
                                  data-button-test={true}
                                  onClick={[MockFunction]}
                                  type="button"
                                >
                                  <button
                                    className="sc-AxheI bgVlsT sc-AxiKw Rtuhc"
                                    color=""
                                    data-button-test={true}
                                    onClick={[MockFunction]}
                                    type="button"
                                  >
                                    Confirm
                                  </button>
                                </styled.button>
                              </styled.button>
                            </ForwardRef>
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </Typography>
                </div>
              </Modal__wrap>
            </div>
          </baseStyles__BaseModalBackground>
        </Portal>
      </j>
    </Component>
  </Confirm>
  <div />
</p>
`;
